### DISCUSSION TITLE
Hide password fields on User creation page

### DISCUSSION THREAD

**Discussion by Graph2133** [OrchardCMS/OrchardCore#12796] [Q&A]
Hello, I migrated Orchard Core from v1.2.2 to 1.5.0 and it looks like user creation page was extended with **required** password and password confirmation fields. Is there any option in Orchard to make these fields optional or better hide/remove them ?
![image](https://user-images.githubusercontent.com/23173060/201192849-5025e489-8670-473e-868c-3cc7b68bc957.png)

**Comment by hishamco** [comment]
Why you need to remove them?

**Reply by Graph2133** [reply]
It's because we are using Azure AD authentication and the Create User page is used only to add new users into the system. I assume that there are 2 possible workarounds:
1. Remove UserDisplayDriver from service collection and register new one removing password/password confirmation fields and validation rules.
2. Override `Admin/User/Create` controller action and view
 
The good thing about them is that I can keep other out of the box actions (edit/list/delete) and only create action will change but I am not sure that I am going in the right direction

**Reply by ns8482e** [reply]
I guess password is optional

**Reply by Graph2133** [reply]
Nope, it's required in version 1.5.0. I forgot to mention that local login is used for system administrator authentication so I can not enable "Use only external provider for login" feature.

**Reply by ns8482e** [reply]
Yes Password should only be required for local login - for external identity - we should be able to create a user without need for password. it would be nice to have a user setting that show/hide password field - @Graph2133  would you like  to suggest a PR?

**Reply by hishamco** [reply]
If you are using AAD all the users will be there, you might need to use authentication instead

**Reply by ns8482e** [reply]
IMO if any of the following option is enabled then password should be optional

![image](https://user-images.githubusercontent.com/23270244/201426606-02dca702-bad5-4559-b24f-7c88cad29dd8.png)

**Reply by ns8482e** [reply]
@hishamco even in case of external identity, create user is useful when you don't want to allow all users from that identity to register but allow only users created by admin.

**Reply by hishamco** [reply]
Password is useful for local accounts, even if the user OAuth the ASP.NET maps these accounts. Again I don't see any issue here
